proto/hypertrace/agent/config/v1/config.proto

```diff
@@ -195,5 +195,24 @@ message Telemetry {
 
     // Whether to capture metrics or not. The metrics will be otel go metrics.
     // See https://github.com/open-telemetry/opentelemetry-go/tree/main/metric
-    google.protobuf.BoolValue metrics_enabled = 2;
+    google.protobuf.BoolValue metrics_enabled = 2 [deprecated = true];
+
+    // Metrics config
+    Metrics metrics = 3;
 }
+
+// Metrics has the config to enable OTEl and System metrics
+message Metrics {
+    // Whether to capture metrics or not. The metrics will be otel go metrics.
+    // See https://github.com/open-telemetry/opentelemetry-go/tree/main/metric
+    google.protobuf.BoolValue enabled = 1;
+
+    // System metrics configuration.
+    SystemMetrics system_metrics = 2;
+}
+
+// SystemMetrics has the config for capturing system metrics.
+message SystemMetrics {
+    // Whether to capture system metrics or not. The system metrics is setup in goagent.
```

Comment:
nit: update the comment to `The system metrics is only supported in goagent as of now`

Reply:
`Whether to capture system metrics or not. The system metrics are setup in each agent`
We do not need to add that it's only supported in goagent, because we can update any agent at any time. With this config, any agent can use it.
We can keep this as part of this PR that during the time of this PR, we were only adding support for goagent.

Reply:
You're right, just wanted to propagate the idea that support needs to be added explicitly